v2/pkg/genruntime/core/errors.go

```diff
@@ -207,8 +213,10 @@ func NewSubscriptionMismatchError(expectedSub string, actualSub string) *Subscri
 	}
 }
 
-var _ error = &SubscriptionMismatch{}
```

Comment:
This particular usage of this pattern is a bit funky

Reply:
Agreed